helix-term/src/commands.rs

```diff
@@ -2655,7 +2655,9 @@ fn delete_selection_impl(cx: &mut Context, op: Operation, yank: YankAction) {
         // yank the selection
         let text = doc.text().slice(..);
         let values: Vec<String> = selection.fragments(text).map(Cow::into_owned).collect();
-        let reg_name = cx.register.unwrap_or('"');
+        let reg_name = cx
+            .register
+            .unwrap_or(cx.editor.config().default_yank_register);
```

Comment:
```suggestion
        let reg_name = cx
            .register
            .unwrap_or_else(|| cx.editor.config.load().default_yank_register);
```

The borrow error here is because `Editor::config` takes a shared borrow of the entire `Editor` while `current!` above takes mutable references of some of the fields and the compiler can't detect that we only care about the `.config` field in `Editor::config`. If we access that field directly here we avoid the borrowing conflict